File: algorithms/python/isSubsequence/isSubsequence.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/is-subsequence/
# Author: Miao Zhang
# Date:   2021-02-07

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        queue = collections.deque(s)
        for c in t:
            if not queue: return True
            if c == queue[0]:
                queue.popleft()
        return not queue

File: algorithms/cpp/isSubsequence/isSubsequence.cpp
// Source: https://leetcode.com/problems/is-subsequence/
// Author: Miao Zhang
// Date:   2021-02-07

class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() == 0) return true;
        int i = 0, j = 0;
        while (j < t.size()) {
            if (s[i] == t[j]) {
                i++;
            }
            j++;
            if (i == s.size()) return true;
        }
        return false;
    }
};
